public/app/plugins/datasource/prometheus/result_transformer.test.ts

```diff
@@ -291,6 +291,245 @@ describe('Prometheus Result Transformer', () => {
+              },
+            ],
+          }),
+          new MutableDataFrame({
+            refId: 'A',
+            fields: [
+              { name: 'Time', type: FieldType.time, values: [6, 5, 4] },
+              {
+                name: 'Value',
+                type: FieldType.number,
+                values: [20, 10, 40],
+                labels: { le: '2', additionalProperty: '20' },
+              },
+            ],
+          }),
+          new MutableDataFrame({
+            refId: 'A',
+            fields: [
+              { name: 'Time', type: FieldType.time, values: [6, 5, 4] },
+              {
+                name: 'Value',
+                type: FieldType.number,
+                values: [30, 10, 60],
+                labels: { le: '+Inf', additionalProperty: '20' },
+              },
+            ],
+          }),
+          // 30
+          new MutableDataFrame({
+            refId: 'A',
+            fields: [
+              { name: 'Time', type: FieldType.time, values: [6, 5, 4] },
+              {
+                name: 'Value',
+                type: FieldType.number,
+                values: [30, 30, 60],
+                labels: { le: '1', additionalProperty: '30' },
+              },
+            ],
+          }),
+          new MutableDataFrame({
+            refId: 'A',
+            fields: [
+              { name: 'Time', type: FieldType.time, values: [6, 5, 4] },
+              {
+                name: 'Value',
+                type: FieldType.number,
+                values: [30, 40, 60],
+                labels: { le: '2', additionalProperty: '30' },
+              },
+            ],
+          }),
+          new MutableDataFrame({
+            refId: 'A',
+            fields: [
+              { name: 'Time', type: FieldType.time, values: [6, 5, 4] },
+              {
+                name: 'Value',
+                type: FieldType.number,
+                values: [40, 40, 60],
+                labels: { le: '+Inf', additionalProperty: '30' },
+              },
+            ],
+          }),
+        ],
+      } as unknown as DataQueryResponse;
+
+      const series = transformV2(response, options, {});
+      expect(series.data[0].fields.length).toEqual(4);
+      expect(series.data[0].fields[1].values.toArray()).toEqual([10, 10, 0]);
+      expect(series.data[0].fields[2].values.toArray()).toEqual([10, 0, 30]);
+      expect(series.data[0].fields[3].values.toArray()).toEqual([10, 0, 10]);
+
+      expect(series.data[1].fields[1].values.toArray()).toEqual([0, 10, 10]);
+      expect(series.data[1].fields[2].values.toArray()).toEqual([20, 0, 30]);
+      expect(series.data[1].fields[3].values.toArray()).toEqual([10, 0, 20]);
+
+      expect(series.data[2].fields[1].values.toArray()).toEqual([30, 30, 60]);
+      expect(series.data[2].fields[2].values.toArray()).toEqual([0, 10, 0]);
+      expect(series.data[2].fields[3].values.toArray()).toEqual([10, 0, 0]);
```

Comment:
wonder if we should merge these into a single frame :thinking: cc @ryantxu 

would be a single frame with fields like:

```
Time
1
1
2
2
+Inf
+Inf
```